src/main/java/com/teentech/hotels/controller/ReservationsController.java

```diff
@@ -46,6 +60,31 @@
         try {
             Reservations reservationToSave = ReservationsConverter.convertFromDtoToEntity(reservation);
 
+            if (!reservationToSave.getEverydayCleaning()) {
+                HotelRoomsPK hotelRoomsPK = new HotelRoomsPK(reservationToSave.getHotelId(), reservationToSave.getRoomNumber());
+                HotelRooms room = hotelRoomsService.findHotelRoomById(hotelRoomsPK);
+
+                int noOfDays = (int) ((reservationToSave.getEndDate().getTime() - reservationToSave.getStartDate().getTime()) / (1000 * 60 * 60 * 24));
```

Comment:
Numarul de zile pentru bauturi se poate calcula mai simplu, folosing clasa Calendar gasesti diferenta in zile dintre 2 date si apoi scazi ziua curenta